rails_app/app/controllers/currencies_controller.rb

```diff
@@ -15,6 +15,18 @@ def index
       redirect_to all_currencies_path
     end
   end
+  
+  # GET /currencies/all or /currencies/all.json
+  def all
+    @currencies = Currency.all
+
+    if user_signed_in?
+      user = current_user
+
+      @followed_codes = user.followed_currencies.map { |curr| curr.code }
```

Comment:
This is iterating all currencies in memory and then fetching the code of each one. I believe it could be done directly with:

```ruby
user.followed_currencies.pluck(:code)
```

Give it a try!

Reference: https://apidock.com/rails/ActiveRecord/Calculations/pluck